lib/acme/resources/challenges/tlssni01.rb

```diff
@@ -0,0 +1,24 @@
+class Acme::Resources::Challenges::TLSSNI01 < Acme::Resources::Challenges::Base
+
+  def hostname
+    hd = crypto.digest.hexdigest(authorization_key)
+    "#{hd[0..31]}.#{hd[32..64]}.acme.invalid"
+  end
+
+  def temporary_cert
+    @temporary_cert ||= ::Acme::Certificate.new(nil, nil, nil).create_self_signed_cert([domain])
```

Comment:
Really, this API is a bit terrible this way and atm nobody will find any use for using `create_self_signed_cert` in another context. Just move it into method of this class that returns an `Acme::Client::Certificate` please.